wordpress-mobile/WordPress-Android
# Issue/12039 reader improvements repository

Addresses #12039 

This PR cleans up some loose ends within the Discover and PostRepository. It includes
- Modified coroutineScope so job is canceled at the Repository level, not at the use case level. Removed `ReaderRepositoryDispatchingUseCase` as a result.
- Use suspend functions in public functions so nothing is accidentally executed on the main thread
- Swapped to use `ReaderPostTableWrapper` and `ReaderPostActionWrapper` in use cases for easier testability and abstraction
- Subscribe to postEnd events on a background thread, so listener code is not inadvertently executed on Main thread
Note: Unit tests will follow shortly

To test
Using the FF set to true
Open WPAndroid 
Navigate to discover
Feed should load

PR submission checklist:

- [] I have considered adding unit tests where possible.
